dynawo/sources/Models/CPP/ModelNetwork/DYNModelBus.cpp

```diff
@@ -315,12 +315,15 @@ ModelBus::evalDerivatives(const double /*cj*/) {
 }
 
 void
-ModelBus::evalF() {
+ModelBus::evalF(propertyF_t type) {
   if (network_->isInitModel()) {
     f_[0] = ir0_;
     f_[1] = ii0_;
   } else {
-    if (getSwitchOff()) {
+    bool switchedOff = getSwitchOff();
+    if (type == DIFFERENTIAL_EQ && (!hasDifferentialVoltages_ || switchedOff))
```

Comment:
You mean skip evaluation of the residuals whenever switchoff is true? Don't we need to compute those at least once?

Reply:
I'll try to clarify: not to test the switchedOff_ argument in the && test with something like -> if (type == DIFFERENTIAL_EQ && !hasDifferentialVoltages_).
During the first time step, type would be different from DIFFERENTIAL_EQ so it would be evaluated. If the bus is then switched off during the simulation, this should be evaluated by the reinit part.

Reply:
ok got it!

Reply:
done